lib/mzgl/midi/AllMidiDevicesAppleImpl.mm

```diff
@@ -335,83 +335,97 @@ static void removeFirst(std::vector<CoreMidiOutRef> &devs, CoreMidiOutRef dev) {
 	}
 }
 
-void AllMidiDevicesAppleImpl::sendPacketList(const MIDIPacketList *packetList) {
+std::optional<MIDIEndpointRef> getEndpointRefForDevice(const MidiDevice &device,
+													   const std::vector<CoreMidiOutRef> &midiOuts) {
 	for (ItemCount index = 0; index < MIDIGetNumberOfDestinations(); ++index) {
-		MIDIEndpointRef outputEndpoint = MIDIGetDestination(index);
-		if (outputEndpoint) {
-			// Send it
-			OSStatus s = MIDISend(outputPort, outputEndpoint, packetList);
-			NSLogError(s, "Sending MIDI");
-		}
+		auto endpoint = MIDIGetDestination(index);
+		if (endpoint == 0) continue;
+
+		for (const auto &destination: midiOuts)
+			if (destination->endpoint == endpoint && device == *destination) return endpoint;
 	}
+
+	return std::nullopt;
```

Comment:
This seemed like the most logical way to get access to the midi destinations, but I might have missed something here - can you take a Quick Look and see if its sensible?